src/ert/gui/tools/plot/plot_api.py

```diff
@@ -205,29 +205,37 @@ def observations_for_key(self, ensemble_ids: List[str], key: str) -> pd.DataFram
                     timeout=self._timeout,
                 )
                 self._check_response(response)
+
+                observations = response.json()
+                observations_dfs = []
+
                 if not response.json():
```

Comment:
Why not replace `if not response.json():` with `if not observations:` if you already declared `observations = response.json()` on ln 209